### ISSUE TITLE
Unable to add an agent due to `undefined method `selected_email_flags='`

### ISSUE THREAD

**Issue by MrSaints** [chatwoot/chatwoot#709]
**Describe the bug**

This is on `chatwoot/chatwoot:v1.3.0`.

When adding an agent, the following exception is triggered:

```
undefined method `selected_email_flags=' for #<NotificationSetting:0x000055bc57eeec88>
Did you mean?  selected_flags=
               select_all_flags
```

And occasionally, it also shows up as:

```
undefined method `conversation_creation?' for #<NotificationSetting:0x0000560ddefc2398>
```

I believe it is related #654. I have followed the recommendations in that issue, but to no avail.

**To Reproduce**

- Start `chatwoot/chatwoot:v1.3.0` server, and sidekiq (both connected to fresh postgresql, and redis)
- Run `rake db:setup` (I have also tried it with this, and `rake db:seed`)
- Login using `john@acme.inc`
- Add an agent at `/app/accounts/1/settings/agents/list`
- Internal Server Error 500

**Expected behavior**

An agent will be added successfully.

**Screenshots**

<img width="840" alt="Screenshot 2020-04-12 at 19 38 33" src="https://user-images.githubusercontent.com/68859/79076896-37030d00-7cf5-11ea-8670-e57c5344739d.png">

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser Google Chrome
 - Version 80.0.3987.163 (Official Build) (64-bit)

**Additional context**

I am running with Docker inside Kubernetes. Below is a snapshot of the command / arguments used:

```yml
          image: chatwoot/chatwoot:v1.3.0
          command:
            - "docker/entrypoints/rails.sh"
          args:
            - "bundle"
            - "exec"
            - "rails"
            - "s"
            - "-p"
            - "3000"
            - "-b"
            - "0.0.0.0"
```

I have attempted adding an agent several times.

Logs below:

```bash
I, [2020-04-12T17:56:17.307569 #1]  INFO -- : [870d10aa-e373-43c6-8631-aee9fb982e08] Completed 200 OK in 81ms (Views: 24.9ms | ActiveRecord: 34.1ms | Allocations: 12659)
I, [2020-04-12T17:56:17.302618 #1]  INFO -- : [2d8cf5bd-9b46-4e46-a7e4-85b2565dffd0]   Rendering api/v1/accounts/show.json.jbuilder
I, [2020-04-12T17:56:17.310753 #1]  INFO -- : [2d8cf5bd-9b46-4e46-a7e4-85b2565dffd0]   Rendered api/v1/accounts/show.json.jbuilder (Duration: 0.7ms | Allocations: 111)
I, [2020-04-12T17:56:17.312888 #1]  INFO -- : [2d8cf5bd-9b46-4e46-a7e4-85b2565dffd0] Completed 200 OK in 56ms (Views: 9.2ms | ActiveRecord: 13.9ms | Allocations: 8675)
I, [2020-04-12T17:58:55.919414 #1]  INFO -- : [5dce6b85-51c0-41ad-8485-159281d31ffd] Started POST "/api/v1/accounts/1/agents" for 10.42.1.20 at 2020-04-12 17:58:55 +0000
I, [2020-04-12T17:58:55.926298 #1]  INFO -- : [5dce6b85-51c0-41ad-8485-159281d31ffd] Processing by Api::V1::Accounts::AgentsController#create as JSON
I, [2020-04-12T17:58:55.926767 #1]  INFO -- : [5dce6b85-51c0-41ad-8485-159281d31ffd]   Parameters: {"name"=>"<omitted>", "email"=>"<omitted>@<omitted>", "role"=>"agent", "account_id"=>"1", "agent"=>{"name"=>"<omitted>", "email"=>"<omitted>@<omitted>", "role"=>"agent"}}
W, [2020-04-12T17:58:55.927986 #1]  WARN -- : [5dce6b85-51c0-41ad-8485-159281d31ffd] Can't verify CSRF token authenticity.
I, [2020-04-12T17:58:55.991976 #1]  INFO -- : [5dce6b85-51c0-41ad-8485-159281d31ffd] Completed 500 Internal Server Error in 65ms (ActiveRecord: 12.8ms | Allocations: 4723)
I, [2020-04-12T17:58:56.002850 #1]  INFO -- : [5dce6b85-51c0-41ad-8485-159281d31ffd] Sending event c243545cd3c849f5914b73344c287c3a to Sentry
F, [2020-04-12T17:58:56.585947 #1] FATAL -- : [5dce6b85-51c0-41ad-8485-159281d31ffd]
[5dce6b85-51c0-41ad-8485-159281d31ffd] NoMethodError (undefined method `selected_email_flags=' for #<NotificationSetting:0x000055bc590ff9b8>
Did you mean?  selected_flags=
               select_all_flags):
[5dce6b85-51c0-41ad-8485-159281d31ffd]
[5dce6b85-51c0-41ad-8485-159281d31ffd] app/models/account_user.rb:40:in `create_notification_setting'
[5dce6b85-51c0-41ad-8485-159281d31ffd] app/controllers/api/v1/accounts/agents_controller.rb:48:in `save_account_user'
[5dce6b85-51c0-41ad-8485-159281d31ffd] app/controllers/application_controller.rb:39:in `handle_with_exception'
I, [2020-04-12T17:59:00.827168 #1]  INFO -- : [13f439c9-9071-4afc-9516-809286bcfb2f] Started POST "/api/v1/accounts/1/agents" for 10.42.1.20 at 2020-04-12 17:59:00 +0000
I, [2020-04-12T17:59:00.831944 #1]  INFO -- : [13f439c9-9071-4afc-9516-809286bcfb2f] Processing by Api::V1::Accounts::AgentsController#create as JSON
I, [2020-04-12T17:59:00.832136 #1]  INFO -- : [13f439c9-9071-4afc-9516-809286bcfb2f]   Parameters: {"name"=>"<omitted>", "email"=>"<omitted>@<omitted>", "role"=>"agent", "account_id"=>"1", "agent"=>{"name"=>"<omitted>", "email"=>"<omitted>@<omitted>", "role"=>"agent"}}
W, [2020-04-12T17:59:00.833294 #1]  WARN -- : [13f439c9-9071-4afc-9516-809286bcfb2f] Can't verify CSRF token authenticity.
I, [2020-04-12T17:59:00.865712 #1]  INFO -- : [13f439c9-9071-4afc-9516-809286bcfb2f] Completed 500 Internal Server Error in 33ms (ActiveRecord: 7.0ms | Allocations: 4719)
I, [2020-04-12T17:59:00.878037 #1]  INFO -- : [13f439c9-9071-4afc-9516-809286bcfb2f] Sending event 0d56237e2c694c18955cbff8ac621afb to Sentry
F, [2020-04-12T17:59:01.466508 #1] FATAL -- : [13f439c9-9071-4afc-9516-809286bcfb2f]
[13f439c9-9071-4afc-9516-809286bcfb2f] NoMethodError (undefined method `selected_email_flags=' for #<NotificationSetting:0x000055bc57eeec88>
Did you mean?  selected_flags=
               select_all_flags):
[13f439c9-9071-4afc-9516-809286bcfb2f]
[13f439c9-9071-4afc-9516-809286bcfb2f] app/models/account_user.rb:40:in `create_notification_setting'
[13f439c9-9071-4afc-9516-809286bcfb2f] app/controllers/api/v1/accounts/agents_controller.rb:48:in `save_account_user'
[13f439c9-9071-4afc-9516-809286bcfb2f] app/controllers/application_controller.rb:39:in `handle_with_exception'
I, [2020-04-12T18:00:10.004270 #1]  INFO -- : [f564f82d-b25e-49a3-9d70-b047cc85d1df] Started POST "/api/v1/accounts/1/agents" for 10.42.1.20 at 2020-04-12 18:00:10 +0000
I, [2020-04-12T18:00:10.012144 #1]  INFO -- : [f564f82d-b25e-49a3-9d70-b047cc85d1df] Processing by Api::V1::Accounts::AgentsController#create as JSON
I, [2020-04-12T18:00:10.012619 #1]  INFO -- : [f564f82d-b25e-49a3-9d70-b047cc85d1df]   Parameters: {"name"=>"<omitted>", "email"=>"<omitted>@<omitted>", "role"=>"agent", "account_id"=>"1", "agent"=>{"name"=>"<omitted>", "email"=>"<omitted>@<omitted>", "role"=>"agent"}}
W, [2020-04-12T18:00:10.014301 #1]  WARN -- : [f564f82d-b25e-49a3-9d70-b047cc85d1df] Can't verify CSRF token authenticity.
I, [2020-04-12T18:00:10.058694 #1]  INFO -- : [f564f82d-b25e-49a3-9d70-b047cc85d1df] Completed 500 Internal Server Error in 46ms (ActiveRecord: 6.2ms | Allocations: 4723)
I, [2020-04-12T18:00:10.099138 #1]  INFO -- : [f564f82d-b25e-49a3-9d70-b047cc85d1df] Sending event 369a46728076452ebc6bc483e01b0039 to Sentry
F, [2020-04-12T18:00:10.679682 #1] FATAL -- : [f564f82d-b25e-49a3-9d70-b047cc85d1df]
[f564f82d-b25e-49a3-9d70-b047cc85d1df] NoMethodError (undefined method `selected_email_flags=' for #<NotificationSetting:0x000055bc5505f5e8>
Did you mean?  selected_flags=
               select_all_flags):
[f564f82d-b25e-49a3-9d70-b047cc85d1df]
[f564f82d-b25e-49a3-9d70-b047cc85d1df] app/models/account_user.rb:40:in `create_notification_setting'
[f564f82d-b25e-49a3-9d70-b047cc85d1df] app/controllers/api/v1/accounts/agents_controller.rb:48:in `save_account_user'
[f564f82d-b25e-49a3-9d70-b047cc85d1df] app/controllers/application_controller.rb:39:in `handle_with_exception'
```

**Comment by sojan-official** [comment]
if you are upgrading an existing installation of chatwoot. could you run the following command?   

```
rake db:migrate
```

this should be done after all the version upgrades, sorry for the lack of documentation on this. we are working on an update instructions documentation.

**Comment by MrSaints** [comment]
Hey @sojan-official ! Thanks for the very prompt response. It's a fresh installation. But I'll give `rake db:migrate` a shot.

**Comment by MrSaints** [comment]
No luck so far.

```
[a08b7e21-3910-484f-a704-db05c3570c6a] NoMethodError (undefined method `selected_email_flags=' for #<NotificationSetting:0x00005562ad68c198>
Did you mean?  selected_flags=
               select_all_flags):
```

This is with `rake db:migrate` alone, and with `bundle exec rails db:setup` followed by `bundle exec rails db:migrate`.

**Comment by sojan-official** [comment]
@MrSaints  Thanks for the update. let me debug this at my end. I will get back for more info if I can't reproduce.

**Comment by sojan-official** [comment]
@MrSaints  I was trying to reproduce this and had a hard time making it happen.  Can we give it one more try from scratch and if it's persisting our team will get on a call to help you out. 

Steps: 
- Delete the existing database 
- let's use  the latest image for setting up : `chatwoot/chatwoot:latest`
- After you have everything in place, run  `rake db:setup` this should take care of the seeding as well. 
- restart the rails and sidekiq pods after running the migration
- try to add a new agent. 

if the issue is persisting, lets sync over https://calendly.com/chatwoot/ at a time that works for you.

**Comment by MrSaints** [comment]
@sojan-official Thank you so much for putting so much time, and effort into helping us. I can confirm that it is now working thanks to your recommendations. 

We have altered our Kubernetes `spec` to look like this:

```
    spec:
      containers:
        - name: app
          image: chatwoot/chatwoot:latest
          command:
            - "bin/rails"
          args:
            - "server"
            - "-p"
            - "3000"
            - "-b"
            - "0.0.0.0"
```

It is also using the `latest` image.

Upon launching it, we ran `rake db:setup`.

We then deleted the pod (essentially restarting).

And it all works now. Thanks again for your assistance. We really appreciate it.
